app/models/workflow.rb

```diff
@@ -5,7 +5,9 @@ class Workflow < ApplicationRecord
   default_scope { order(:sequence => :asc) }
 
   belongs_to :template
-  has_many :requests, -> { order(:id => :asc) }, :inverse_of => :workflow
+  before_destroy :validate_deletable, :prepend => true
+  after_destroy :send_deletion_message
```

Comment:
should change to `after_commit`

Reply:
yes, make sure the message is only broadcasted after AR is really deleted